README.md

```diff
@@ -1,27 +1,67 @@
+when contracts are transacted on by untrusted parties. 
+By providing examples of commit-reveal implementations we hope to foster a more secure ecosystem of decentralized
+applications and encourage developers to build with best practices.
+
+## Commit-Reveal Scheme
+
+The contracts contained in this repo demonstrate how to use Flow's onchain randomness safely
+in contracts that are transacted on by untrusted parties. Safe randomness here meaning non-revertible randomness, 
+i.e. mitigating post-selection attacks via a commit-and-reveal scheme.
 
 Random sources are committed to the [`RandomBeaconHistory` contract](./contracts/RandomBeaconHistory.cdc) by the service
-account at the end of every block. These random sources are catalogued chronologically, extending historically for every
-associated block height to the initial commitment height.
+account at the end of every block. The RandomBeaconHistory contract provides a convenient archive, where for each past
+block height (starting Nov 2023) the respective 'source of randomness' can be retrieved.
 
-Used on their own, these random sources are not safe. In other words, using the random source in your contract without
-the framing of a commit-reveal mechanism would enable non-trusted callers to condition their interactions with your contract on the
+When used naively, `revertibleRandom` as well as the [`RandomBeaconHistory` contract](./contracts/RandomBeaconHistory.cdc)
+are subject to post-selection attacks from transactions.
+In simple terms, using the random source in your contract without
+the protection of a commit-reveal mechanism would enable non-trusted callers to condition their interactions with your contract on the
 random result. In the context of a random coin toss, I could revert my transaction if I didn't win - not a very fair
 game.
 
 To achieve non-revertible randomness, the contract should be structured to resolve in two phases:
 
 1. Commit - Caller commits to the resolution of their bet with some yet unknown source of randomness (i.e. in the
    future)
-2. Reveal - Caller can then reveal the result of their bet
+2. Reveal - Caller can then resolve the result of their bet once the source of randomness is available in the `RandomBeaconHistory` with a separate transaction.
+   From a technical perspective, this could also be called "resolving transaction", because the transaction simply executes the smart contract with the locked-in
+   inputs, whose output all parties committed to accept in the previous phase.
 
-Though a caller could still condition the revealing transaction on the coin flip result, they've already incurred the
-cost of their bet and would gain nothing by doing so.
+Though a caller could still condition the revealing transaction on the coin flip result, all the inputs influencing the bet's outcome
+have already been fixed (the source of randomness being the last one that is only generated after the commit transaction concluded.)
```

Comment:
```suggestion
have already been fixed (the source of randomness being the last one that is only generated after the commit transaction concluded).
```

Reply:
applied in https://github.com/onflow/random-coin-toss/pull/9/commits/5499612d6aa8fb8138c16f0486484c2793a5a722